instrumentation/github.com/aws/aws-lambda-go/otellambda/lambda.go

```diff
@@ -0,0 +1,158 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otellambda
+
+import (
+	"context"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/lambdacontext"
+
+	"go.opentelemetry.io/contrib/propagators/aws/xray"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	tracerName = "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda"
+)
+
+var errorLogger = log.New(log.Writer(), "OTel Lambda Error: ", 0)
+
+type Flusher interface {
+	ForceFlush(context.Context) error
+}
+
+type noopFlusher struct{}
+
+func (*noopFlusher) ForceFlush(context.Context) error { return nil }
+
+// Compile time check our noopFlusher implements FLusher
+var _ Flusher = &noopFlusher{}
+
+type InstrumentationOption func(o *InstrumentationOptions)
+
+type InstrumentationOptions struct {
+	// TracerProvider is the TracerProvider which will be used
+	// to create instrumentation spans
+	// The default value of TracerProvider the global otel TracerProvider
+	// returned by otel.GetTracerProvider()
+	TracerProvider trace.TracerProvider
+
+	// Flusher is the mechanism used to flush any unexported spans
+	// each Lambda Invocation to avoid spans being unexported for long
+	// when periods of time if Lambda freezes the execution environment
+	// The default value of Flusher is a noop Flusher, using this
+	// default can result in long data delays in asynchronous settings
+	Flusher Flusher
+}
+
+var configuration InstrumentationOptions
+var resourceAttributesToAddAsSpanAttributes []attribute.KeyValue
+var tracer trace.Tracer
```

Comment:
These should be encapsulated in a struct and not left as package global variables.  The free functions in this package that operate on them should probably then be methods with the new struct that contains them as a receiver.